cmd/icingadb-migrate/convert.go

```diff
@@ -224,7 +224,7 @@ type downtimeRow = struct {
 	CommentData         string
 	IsFixed             uint8
 	Duration            int64
-	ScheduledStartTime  int64
+	ScheduledStartTime  sql.NullInt64
 	ScheduledEndTime    int64
 	ActualStartTime     int64
```

Comment:
According to https://github.com/Icinga/icingadb/issues/550#issuecomment-1348690659, `actual_start_time` also was `NULL`. Might be worth looking at the schema which columns are allowed to be `NULL` an handle all these.

Reply:
Isn't worth it IMAO. Everything that has a reason to be NULL is already handled.

Reply:
"a reason to be NULL" may also be "there was an old icinga2 version that had a bug that didn't populate that column"

Reply:
\*rolling eyes\* s/reason/actual reason/

I'd rather wait for that bug also arriving here.

Reply:
> Isn't worth it IMAO. Everything that has a reason to be NULL is already handled.

It's worth it. This bug report is proof of that.

> I'd rather wait for that bug also arriving here.

I would rather fix them before. We know that almost everyone had or has an Icinga 2 version that has bugs 😆.

Reply:
I'm afraid there _could_ be NULL literally everywhere ex. IDs.

```
$ grep -Fwie 'NOT NULL' <lib/db_ido_mysql/schema/mysql.sql |grep -vFe _id
$
```

On the other hand there's only 1 report yet.

Still sure about the effort/gain ratio?